Bottom: 7a3ca88d1c9bb774d09c4439ed22e5c0ea628e2d
Top:    00a967dcebdc90201a74ff1c24e06c5a1956067c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-23 19:27:04 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 69259f5..0766b26 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
